mlir/CMakeLists.txt

```diff
@@ -1,3 +1,16 @@
+if(NOT DEFINED LLVM_VERSION_MAJOR)
+  set(LLVM_VERSION_MAJOR 16)
+endif()
+if(NOT DEFINED LLVM_VERSION_MINOR)
+  set(LLVM_VERSION_MINOR 0)
+endif()
+if(NOT DEFINED LLVM_VERSION_PATCH)
+  set(LLVM_VERSION_PATCH 0)
```

Comment:
Ok, actual request, having stared at this all some. We should define the patch version to, for ROCm 5.5, `50500`, to match the amdgpu drivers' conventions for these things. That is, `[0]50500` for `5.5.0`)